src/codegen/mod.rs

```diff
@@ -3962,6 +3964,53 @@ impl CodeGenerator for ObjCInterface {
                         }
                     };
                     result.push(impl_trait);
```

Comment:
This is getting kinda gnarly, it may be worth factoring this `if let ... { .. } else { .. }` to its own function. Follow-up is fine.